contracts/farm/LiquidityMining.sol

```diff
@@ -731,4 +764,65 @@ contract LiquidityMining is ILiquidityMining, LiquidityMiningStorage {
         UserInfo memory ui = getUserInfo(_poolToken, _user);
         return ui.amount;
     }
+
+    /**
+     * @notice returns the accumulated liquid reward for the given user for each pool token
+     * @param _user the address of the user
+     */
+    function getUserAccumulatedLiquidReward(address _user)
+        external
+        view
+        returns (uint256[] memory)
+    {
+        uint256 length = poolInfoList.length;
+        uint256[] memory rewardList = new uint256[](length);
+        for (uint256 i = 0; i < length; i++) {
+            address _poolToken = address(poolInfoList[i].poolToken);
+            uint256 _unlockedImmediatelyPercent =
+                _getPoolTokenUnlockedImmediatelyPercent(_poolToken);
+            rewardList[i] =
+                (_unlockedImmediatelyPercent * _getUserAccumulatedReward(i, _user)) /
+                10000;
+        }
+
+        return rewardList;
+    }
+
+    /**
+     * @notice returns the accumulated vested reward for the given user for each pool token
+     * @param _user the address of the user
+     */
+    function getUserAccumulatedVestedReward(address _user)
+        external
+        view
+        returns (uint256[] memory)
+    {
+        uint256 length = poolInfoList.length;
+        uint256[] memory rewardList = new uint256[](length);
+        for (uint256 i = 0; i < length; i++) {
+            address _poolToken = address(poolInfoList[i].poolToken);
+            uint256 _unlockedImmediatelyPercent =
+                _getPoolTokenUnlockedImmediatelyPercent(_poolToken);
+            rewardList[i] =
+                ((10000 - _unlockedImmediatelyPercent) * _getUserAccumulatedReward(i, _user)) /
+                10000;
+        }
+
+        return rewardList;
```

Comment:
same here

Reply:
updated in this [commit](https://github.com/DistributedCollective/Sovryn-smart-contracts/pull/493/commits/1653d961d13cf23ed8973583efdbc8db0e26194e)